### PR TITLE
Make splash occur only on harm intent

### LINKED ISSUE #2546: Harm Intent to Splash Beakers
Gray: SUGG: Make it so you need to be on harm intent to splash beakers on stuff.

### PR DESCRIPTION
**Pull request by hitsedesen**
closes #2546

### CODE REVIEW

**Comment by LagannPM** [comment]
What about sink interaction ?

**Comment by hitsedesen** [comment]
image

**Comment by LagannPM** [comment]
it should be "into the sink" instead.

**Comment by hitsedesen** [comment]
Why so?

**Comment by LagannPM** [comment]
and reagents should be just removed if you dispose them into sink, as they are disposed safely.
Otherwise you just splash content of the beaker all over the sink.

File: code/modules/reagents/reagent_containers/glass.dm [hunk 1]
```diff
@@ -108,6 +97,17 @@
 		else
 			name = "[base_name] ([label_text])"
 
+/obj/item/weapon/reagent_containers/glass/beaker/pre_attack(atom/a, mob/user, params)
+	if(user.a_intent == I_HURT)
+		if(standard_splash_mob(user, a))
+			return 1
```
Comment (seekin4u) [inline code/modules/reagents/reagent_containers/glass.dm]: Code convention says to use TRUE/FALSE instead of 1/0

**Comment by NanakoAC** [comment]
Please clarify the state of sink interaction now

**Comment by hitsedesen** [comment]
`splash onto`, as above

**Comment by EvenInDeathIStillServe** [comment]
It should be "You pour the solution down the sink."
Because that's what you do, you don't splash water on a sink: it's just going to get the floor wet. You deliberately pour it into the sink, and it goes down the drain.

**Comment by NanakoAC** [comment]
change it to pouring chemicals down the sink please

**Pull request merged**